packages/block-editor/src/components/rich-text/index.native.js

```diff
@@ -103,22 +105,33 @@ export class RichText extends Component {
 		this.onPaste = this.onPaste.bind( this );
 		this.onFocus = this.onFocus.bind( this );
 		this.onBlur = this.onBlur.bind( this );
+		this.onTextUpdate = this.onTextUpdate.bind( this );
 		this.onContentSizeChange = this.onContentSizeChange.bind( this );
 		this.onFormatChangeForceChild = this.onFormatChangeForceChild.bind( this );
 		this.onFormatChange = this.onFormatChange.bind( this );
+		this.formatToValue = memize(
+			this.formatToValue.bind( this ),
+			{ maxSize: 1 }
+		);
+
 		// This prevents a bug in Aztec which triggers onSelectionChange twice on format change
 		this.onSelectionChange = this.onSelectionChange.bind( this );
+		this.onSelectionChangeFromAztec = this.onSelectionChangeFromAztec.bind( this );
 		this.valueToFormat = this.valueToFormat.bind( this );
 		this.willTrimSpaces = this.willTrimSpaces.bind( this );
 		this.state = {
-			start: 0,
-			end: 0,
-			formatPlaceholder: null,
+			activeFormats: [],
```

Comment:
Those updates (`formatPlaceholder`=>`activeFormats`, `this.lastContent` => `this.value`, `this.formatToValue`...) aren't required per se but they bring the code closer to the web version